docker/docker_nightly.py

```diff
@@ -1,26 +1,52 @@
 from datetime import date
 import os
+import sys
+import argparse
 
 def get_nightly_version():
     today = date.today()
     return today.strftime("%Y.%m.%d")
 
 if __name__ == "__main__":
+
+    def try_and_handle(cmd, dry_run = False):
+        if dry_run:
+            print(f"Executing command: {cmd}")
+        else:
+            code = os.system(cmd)
```

Comment:
will os.system return error if docker build fails? Should we use subprocess.run as https://github.com/pytorch/serve/pull/1623/files#diff-0377cd7efa39189d481ea81cd7c45057e164e01393c2316560bc6bf677b21517R17

Reply:
`os.system(cmd)` will not return an error but it'll return a non zero error code if it fails and get added to a `failed_commands` list 

Then at the very end I raise an exception so the script will definitely fail and you can see this in the logs above in the description

```python
    if failed_commands:
        raise Exception(f"failed commands: {failed_commands}")
```

That said if you think `subprocess.run` is a better long term option I can use, `os.system()` is just easy

Reply:
Updated to subprocess.run here https://github.com/pytorch/serve/pull/1627#issuecomment-1129467968